x/pki/keeper/msg_server_revoke_x_509_cert.go

```diff
@@ -49,5 +49,8 @@ func (k msgServer) RevokeX509Cert(goCtx context.Context, msg *types.MsgRevokeX50
 	// remove from subject -> subject key ID map
 	k.RemoveApprovedCertificateBySubject(ctx, msg.Subject, msg.SubjectKeyId)
 
+	// remove from subject key ID -> certificates map
+	k.RemoveApprovedCertificatesBySubjectKeyID(ctx, msg.SubjectKeyId)
```

Comment:
What if we have multiple certificates with the same SubjectKeyId, but different subjects? It seems they all will be removed, which is not correct (here and in other similar places).